src/mui/button/RefreshButton.js

```diff
@@ -2,26 +2,37 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import compose from 'recompose/compose';
+import withHandlers from 'recompose/withHandlers';
 import FlatButton from 'material-ui/FlatButton';
 import NavigationRefresh from 'material-ui/svg-icons/navigation/refresh';
 import translate from '../../i18n/translate';
 import { refreshList as refreshListAction } from '../../actions/listActions';
 
-const RefreshButton = ({ label = 'aor.action.refresh', translate, refresh, resource }) => <FlatButton
+const RefreshButton = ({ label, translate, onClick }) => <FlatButton
     primary
     label={label && translate(label)}
-    onClick={() => refresh(resource)}
+    onClick={onClick}
     icon={<NavigationRefresh />}
 />;
 
+RefreshButton.defaultProps = {
+    label: 'aor.action.refresh',
+};
+
 RefreshButton.propTypes = {
     label: PropTypes.string,
-    refresh: PropTypes.func.isRequired,
+    onClick: PropTypes.func.isRequired,
     translate: PropTypes.func.isRequired,
     resource: PropTypes.string.isRequired,
 };
 const enhance = compose(
     connect(null, { refresh: refreshListAction }),
```

Comment:
Sorry about previous request. We don't need recompose `withHandlers` for that as we are already using `connect`:
```js
connect(null, (dispatch, { source }) => ({
    refresh: (event) => {
        event.preventDefault();
        dispatch(refreshListAction(source));
    },
}));
```